internal/provider/data_file.go

```diff
@@ -0,0 +1,121 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/neuspaces/terraform-provider-system/internal/client"
+	"path"
+)
+
+const dataFileName = "system_file"
+
+func dataFile() *schema.Resource {
+	return &schema.Resource{
+		Description: fmt.Sprintf("`%s` retrieves information about a file on the remote system.", dataFileName),
+
+		ReadContext: dataFileRead,
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			resourceFileAttrId: {
```

Comment:
Please define all attribute names of the data source as separate constants and do not reuse the `resourceFileAttr...`. Although the attribute names mostly overlap with the file resource, The data source should define its own `dataFileAttrId` instead of using the `resourceFileAttrId`. Refer to the other data sources as an example. The intention of these constants is to make use of IDE syntax features, e.g. find usages or a consistently rename attributes.

```suggestion
			dataFileAttrId: {
```

Reply:
ok, I used resourceFileAttr... to set dataFileAttr for consistent naming